zerver/lib/user_groups.py

```diff
@@ -730,11 +732,9 @@ def parse_group_setting_value(
     if len(setting_value.direct_members) == 0 and len(setting_value.direct_subgroups) == 1:
         return setting_value.direct_subgroups[0]
 
-    if not settings.ALLOW_ANONYMOUS_GROUP_VALUED_SETTINGS:
+    if not settings.ALLOW_GROUP_VALUED_SETTINGS:
         raise JsonableError(
-            _("{setting_name} can only be set to a single named user group.").format(
-                setting_name=setting_name
-            )
+            _("'{setting_name}' must be a system user group.").format(setting_name=setting_name)
```

Comment:
Updated the error message to be same as what we have in `require_system_group` check.

Reply:
OK, this is fine but the cleaner way to do that is a new subclass in `zerver/lib/exceptions.py`, and then both code paths can just `raise SystemGroupRequiredError(setting_name)` or whatever.